pkg/config/profile.go

```diff
@@ -45,10 +40,15 @@ func (p *Profile) GetDeviceName() (string, error) {
 	return "", errors.New("your device name has not been configured. Use `stripe login` to set your device name")
 }
 
-// GetSecretKey will return the existing key for the given profile
-func (p *Profile) GetSecretKey() (string, error) {
-	// Try to fetch the API key from the command-line flag or the environment first
-	key := viper.GetString("secret_key")
+// GetAPIKey will return the existing key for the given profile
+func (p *Profile) GetAPIKey() (string, error) {
+	// If the user doesn't have an api_key field set, they might be using an
+	// old configuration so try to read from secret_key
+	if !viper.IsSet(p.GetConfigField("api_key")) {
+		p.RegisterAlias("api_key", "secret_key")
+	}
+
+	key := viper.GetString(p.GetConfigField("api_key"))
```

Comment:
I think this is incorrect, it should be:
```
key := viper.GetString("api_key")
```

Otherwise you're only reading using the profile prefix, which we're already doing below (L62).